src/app/policies/sources.ts

```diff
@@ -0,0 +1,44 @@
+import { DataSourceResponse } from '@/app/application/services/data-source/DataSource'
+import type KumaApi from '@/services/kuma-api/KumaApi'
+import type {
+  PolicyEntity as Policy,
+  PolicyType,
+} from '@/types/index.d'
+export interface CollectionResponse<T = Record<string, unknown>> {
+  total: number,
+  items: T[]
+  next: string | null
+}
+
+type MeshParams = {
+  mesh: string
+}
+type PaginationParams = {
+  offset: number
+}
+
+type PolicyTypeParams = {
+  policyType: string
+}
+// type PolicyParams = {
+//   policy: string
+// }
```

Comment:
**Suggestion**: Please move types like this into `src/types/api.d.ts`.

Reply:
Yeah for sure. Certain things will be moved as part of the WIP/before the end of the WIP. But certain single use types won't be. We also need to consider the modular nature of things. Eventually lot of this will be autogenerated into a specific place anyway so we don't have to hand write them.

Reply:
See https://github.com/kumahq/kuma-gui/pull/1132 for the removal of the `null` on `items`

Reply:
I mostly want to avoid the common "now we have two places for this" problem. We already have a central place for this that adequately covers our needs for storage of types like this.

I also really disagree that these types should live here at all. Our APIs may be called from arbitrary points of our application and so arbitrary places may want to reference the parameter types (etc.).

Reply:
The `sources` files will become the central place for anything to do with HTTP. This should not leak through in to the application code as it currently does, therefore we shouldn't need these types anywhere but here. Moreover here when I say 'central' I mean central to the module, not the application.

Take into account I pulled back a lot of this to keep using KumaApi _temporarily_. This PR is a path to removing all of that and ensuring there is a good boundary between the data layer and the view layer i.e.:

> Our APIs may be called from arbitrary points of our application and so arbitrary places may want to reference the parameter types

We should avoid using the HTTP parameter types in arbitrary places in the application.